src/include/duckdb/common/types.hpp

```diff
@@ -394,8 +394,6 @@ struct LogicalType {
 	DUCKDB_API static LogicalType MAP(LogicalType key, LogicalType value);       // NOLINT
 	DUCKDB_API static LogicalType UNION(child_list_t<LogicalType> members);      // NOLINT
 	DUCKDB_API static LogicalType ARRAY(const LogicalType &child, idx_t size);   // NOLINT
```

Comment:
Can we turn this into an `optional_idx` instead to avoid having to use `DConstants::INVALID_INDEX` as a magic number?